### ISSUE TITLE
`D102` & `D103` should not warn on functions where all overloads have docstrings

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#8084]
```py
from typing import overload

@overload
def foo(a: int) -> None:
    """asdf"""

@overload
def foo(a: str) -> None:
    """fdsa"""

def foo(a: object) -> None: # D103: Missing docstring in public function
    ...
```
https://play.ruff.rs/20d01228-43dd-4364-a17d-c2a936781942

in vscode, the docstring that shows on hover is different depending on the overload being used, so i often use overload-specific documentation instead of documenting the implementation

**Comment by charliermarsh** [comment]
Hmm, this doesn't seem to be true in PyCharm, and I don't _think_ Sphinx supports it either. It seems like documenting the non-overloaded implementation is the best practice?

**Comment by DetachHead** [comment]
perhaps it could be a setting?
